src/common/core/core.js

```diff
@@ -3911,7 +3942,406 @@ define([
+         * @param {string} newSetName - the name of the target set where the member should be moved to.
+         *
+         * @throws {CoreIllegalArgumentError} If some of the parameters doesn't match the input criteria.
+         * @throws {CoreIllegalOperationError} If the context of the operation is not allowed.
+         * @throws {CoreAssertError} If some internal error took place inside the core layers.
+         */
+        this.moveMember = function (node, memberPath, oldSetName, newSetName) {
+            ensureNode(node, 'node');
+            ensureType(memberPath, 'memberPath', 'string');
+            ensurePath(memberPath, 'memberPath');
+            ensureType(oldSetName, 'oldSetName', 'string');
+            ensureType(newSetName, 'newSetName', 'string');
+
+            if (core.getSetNames(node).indexOf(oldSetName) === -1) {
+                throw new CoreIllegalOperationError('Source set [' + oldSetName + '] does not exists.');
+            }
+
+            if (core.getSetNames(node).indexOf(newSetName) === -1) {
+                throw new CoreIllegalOperationError('Target set [' + newSetName + '] does not exists.');
+            }
+
+            if (core.getOwnMemberPaths(node, oldSetName).indexOf(memberPath) === -1) {
+                throw new CoreIllegalOperationError('Not own member of the set therefore cannot be moved.');
+            }
+
+            return core.moveMember(node, memberPath, oldSetName, newSetName);
+        };
+
+        /**
+         * Renames the given attribute definition of the node. It also renames the default value of the definition!
+         * @param {module:Core~Node} node - the node in question.
+         * @param {string} oldName - the current name of the attribute definition in question.
+         * @param {string} newName - the new name of the attribute.
+         *
+         * @throws {CoreIllegalArgumentError} If some of the parameters doesn't match the input criteria.
+         * @throws {CoreIllegalOperationError} If the context of the operation is not allowed.
+         * @throws {CoreAssertError} If some internal error took place inside the core layers.
+         */
+        this.renameAttributeMeta = function (node, oldName, newName) {
+            ensureNode(node, 'node');
+            ensureType(oldName, 'oldName', 'string');
+            ensureType(newName, 'newName', 'string');
+
+            if (core.getValidAttributeNames(node).indexOf(oldName) === -1) {
+                throw new CoreIllegalOperationError('Unknown definition [' + oldName + '] cannot be renamed.');
+            }
+
+            return core.renameAttributeMeta(node, oldName, newName);
+        };
+
+        /**
+         * Moves the given target definition over to a new pointer or set. As actual values in case of
+         * relation definitions vary quite a bit from the meta-targets, this function does not deals with
+         * the actual pointer/set target/members.
```

Comment:
`As actual values in case of relation definitions vary quite a bit from the meta-targets, this function does not deals with the actual pointer/set target/members.`
->
`Note this does not alter the actual pointer target or set members.`